### PR TITLE
add acos to algorithms::math

### PR DESCRIPTION
**Pull request by Heikman**
add arcus cosinus to algorithms::math

### CODE REVIEW

File: src/libPMacc/include/algorithms/math/doubleMath/trigo.tpp [hunk 1]
```diff
@@ -57,6 +57,17 @@ struct Cos<double>
 };
 
 template<>
+struct ACos<double>
+{
+    typedef double result;
+
+    HDINLINE double operator( )(const double& value )
+    {
+        return ::acos( value );
```
Comment (psychocoderHPC) [inline src/libPMacc/include/algorithms/math/doubleMath/trigo.tpp]: On the host side `acosf` is not avail in [C++98 only in c99](http://www.cplusplus.com/reference/cmath/acos/). Please use the overloaded function `acos` on the host side.

**Comment by psychocoderHPC** [comment]
@Heikman please see the last comment

File: src/libPMacc/include/algorithms/math/doubleMath/trigo.tpp [hunk 2]
```diff
@@ -57,6 +57,17 @@ struct Cos<double>
 };
 
 template<>
+struct ACos<double>
+{
+    typedef double result;
+
+    HDINLINE double operator( )(const double& value )
```
Comment (psychocoderHPC) [inline src/libPMacc/include/algorithms/math/doubleMath/trigo.tpp]: Space is missing after the parentheses. `HDINLINE double operator( )( const....)`

File: src/libPMacc/include/algorithms/math/floatMath/trigo.tpp [hunk 3]
```diff
@@ -58,6 +58,21 @@ struct Cos<float>
 };
 
 template<>
+struct ACos<float>
+{
+    typedef float result;
+
+    HDINLINE float operator( )(const float& value )
```
Comment (psychocoderHPC) [inline src/libPMacc/include/algorithms/math/floatMath/trigo.tpp]: Space is missing after the parentheses. `HDINLINE float operator( )( const....)`

File: src/libPMacc/include/algorithms/math/doubleMath/trigo.tpp (hunk 1 above)
Comment (Heikman) [inline src/libPMacc/include/algorithms/math/doubleMath/trigo.tpp]: we are in `doubleMath` :)

**Pull request merged**